spec/unit/typesafe/config/conf_parser_spec.rb

```diff
@@ -781,20 +781,26 @@ def assert_comments_at_path_index(comments, conf, path, index)
   # Skipping 'includeURLHeuristically' because we don't support URLs
   # Skipping 'includeURLBasenameHeuristically' because we don't support URLs
 
-  it "acceptBOMStartingFile" do
-    skip("BOM not parsing properly yet; not fixing this now because it most likely only affects windows") do
-      # BOM at start of file should be ignored
-      conf = Hocon::ConfigFactory.parse_file(TestUtils.resource_file("bom.conf"))
-      expect(conf.get_string("foo")).to eq("bar")
-    end
+  it "doesnotacceptUTF8FileNames" do
+    expect { Hocon::ConfigFactory.parse_file(TestUtils.resource_file("ᚠᛇᚻ.conf")) }.to raise_error
   end
```

Comment:
That's the point of pending: you can put in the behaviour your want, and a
comment on why it is pending, and it will expect a failure on the behaviour
and will notify you once it is *actually* passing. So the test change when
the feature gets implemented is only to remove the pending annotation.

On 14 July 2016 at 16:54, Chris Price <notifications@github.com> wrote:

> In spec/unit/typesafe/config/conf_parser_spec.rb
> <https://github.com/puppetlabs/ruby-hocon/pull/82#discussion_r70831571>:
>
> > @@ -781,20 +781,26 @@ def assert_comments_at_path_index(comments, conf, path, index)
> >    # Skipping 'includeURLHeuristically' because we don't support URLs
> >    # Skipping 'includeURLBasenameHeuristically' because we don't support URLs
> >
> > -  it "acceptBOMStartingFile" do
> > -    skip("BOM not parsing properly yet; not fixing this now because it most likely only affects windows") do
> > -      # BOM at start of file should be ignored
> > -      conf = Hocon::ConfigFactory.parse_file(TestUtils.resource_file("bom.conf"))
> > -      expect(conf.get_string("foo")).to eq("bar")
> > -    end
> > +  it "doesnotacceptUTF8FileNames" do
> > +    expect { Hocon::ConfigFactory.parse_file(TestUtils.resource_file("ᚠᛇᚻ.conf")) }.to raise_error
> >    end
>
> I don't have a strong opinion; your logic makes sense. I think I might be
> inclined to at least throw a comment in here, though, explaining that this
> is not the desired behavior. In the absence of that, in my mind, this test
> reads like an assertion that we don't *want* to support UTF8 file names.
> It's not a huge deal, though.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/puppetlabs/ruby-hocon/pull/82/files/31c2a6774e3ba091d332e070502c68f6183cae94#r70831571>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABsQGkn5g2IgT_HPU7qeD5KLsSnS_kZks5qVluwgaJpZM4JINYw>
> .
>